include/storage/shared_memory.hpp

```diff
@@ -30,6 +30,12 @@ namespace osrm
 namespace storage
 {
 
+enum class Ownership
```

Comment:
Isn't it the case that `shared_memory.hpp` will be included one way or the other anyway? Or are all cpp files compiled separately? (just for my understanding)
i.e. `edge_based_graph` now doesn't depend on `shared_memory` directly, but depends on `static_graph`, which depends on `shared_memory_vector_wrapper`, ... and that one inevitably points to the actual `shared_memory` header file.